parser/src/cfg/mod.rs

```diff
@@ -1642,30 +1644,37 @@ fn parse_action_list(ac: &[SExpr], s: &ParserState) -> Result<&'static KanataAct
 }
 
 fn parse_layer_base(ac_params: &[SExpr], s: &ParserState) -> Result<&'static KanataAction> {
-    Ok(s.a.sref(Action::DefaultLayer(layer_idx(ac_params, &s.layer_idxs)?)))
+    Ok(s.a.sref(Action::DefaultLayer(layer_idx(
+        ac_params,
+        &s.layer_idxs,
+        s,
+    )?)))
 }
 
 fn parse_layer_toggle(ac_params: &[SExpr], s: &ParserState) -> Result<&'static KanataAction> {
-    Ok(s.a.sref(Action::Layer(layer_idx(ac_params, &s.layer_idxs)?)))
+    Ok(s.a.sref(Action::Layer(layer_idx(ac_params, &s.layer_idxs, s)?)))
 }
 
-fn layer_idx(ac_params: &[SExpr], layers: &LayerIndexes) -> Result<usize> {
+fn layer_idx(ac_params: &[SExpr], layers: &LayerIndexes, s: &ParserState) -> Result<usize> {
     if ac_params.len() != 1 {
         bail!(
             "Layer actions expect one item: the layer name, found {} items",
             ac_params.len()
         )
     }
     let layer_name = match &ac_params[0] {
```

Comment:
Seems like this should be turned into something like:

```rust
    let layer_name = ac_params[0].atom(s.vars())
        ...
        .ok_or_else(...)
```

There should be examples of this pattern elsewhere in this file